app/src/main/java/ca/bc/gov/bchealth/widget/AddCommentLayout.kt

```diff
@@ -0,0 +1,75 @@
+package ca.bc.gov.bchealth.widget
+
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import ca.bc.gov.bchealth.R
+import ca.bc.gov.bchealth.databinding.AddCommentBinding
+import ca.bc.gov.bchealth.utils.hideKeyboard
+import ca.bc.gov.bchealth.utils.updateCommentEndIcon
+
+/*
+* Created by amit_metri on 24,June,2022
+*/
+class AddCommentLayout @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private var callback: AddCommentCallback? = null
+    private val binding = AddCommentBinding.inflate(LayoutInflater.from(context), this, true)
+
+    init {
+        binding.tipComment.apply {
+            updateCommentEndIcon(context)
+            setEndIconOnClickListener {
+                if (!binding.edComment.text.isNullOrBlank()) {
+                    callback?.onSubmitComment(binding.edComment.text.toString())
+                    context.hideKeyboard(this)
+                }
+            }
+        }
+
+        binding.edComment.addTextChangedListener(
+            object : TextWatcher {
+                override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                    if (p3 >= 1000) {
```

Comment:
replace p1, p2 and p3 with meaningful variables

Reply:
Updated